compiler-rt/lib/asan/asan_allocator.cpp

```diff
@@ -1007,13 +1007,8 @@ void PrintInternalAllocatorStats() {
   instance.PrintStats();
 }
 
-void asan_free(void *ptr, BufferedStackTrace *stack, AllocType alloc_type) {
```

Comment:
I'm not sure I fully follow - why is it important to no longer expose this `alloc_type` parameter?

Reply:
Its mostly for consistency and ease of reading, especially with free_sized and free_aligned_sized incoming. For every public method related to allocation/deallocation we intercept, we expose exactly one function of the same name in `__asan` with `asan_` as the prefix. Right now memalign is used for all the AllocType and we are about to introduce FROM_ALIGNED_ALLOC. So we have an oddity where we have a 1:1, except for memalign which is special for some reason. The refacator moves the assignment of AllocType to be purely a detail of asan_allocator.cpp.

Its most just to simplify reading and ensure callers do not accidently mix up AllocType.

Reply:
Agreed, it's definitely an improvement in the ergonomics of the API. I'm mostly trying to understand the memalign "specialness", but I'm just asking for context so please don't let me hold this back.